minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntimeDependencies.java

```diff
@@ -0,0 +1,136 @@
+                    .testImplementation(micronautAwsDependency(ARTIFACT_ID_MICRONAUT_AWS_API_PROXY_TEST));
+            if (isLambdaProvided(runtime)) {
+                builder = builder.implementation(micronautAwsDependency(ARTIFACT_ID_MICRONAUT_AWS_CUSTOM_RUNTIME));
+            }
+            return builder.build();
+
+        } else if(runtime == MicronautRuntime.ORACLE_FUNCTION) {
+            return Dependencies.builder()
+                    .implementation(micronautOracleDependency(ARTIFACT_ID_MICRONAUT_ORACLE_HTTP))
+                    .testImplementation(micronautOracleDependency(ARTIFACT_ID_MICRONAUT_ORACLE_HTTP_TEST))
+                    .developmentOnly(micronautOracleDependency(ARTIFACT_ID_MICRONAUT_ORACLE_HTTP_TEST))
+                    .runtimeOnly("com.fnproject.fn:runtime")
+                    .build();
+        } else if(runtime == MicronautRuntime.GOOGLE_FUNCTION) {
+            return Dependencies.builder()
+                    .implementation(micronautGcpDependency(ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP))
+                    .developmentOnly(dependency(GROUP_GOOGLE_CLOUD_FUNCTIONS, ARTIFACT_ID_FUNCTIONS_FRAMEWORK_API))
+                    .developmentOnly(micronautGcpDependency(ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP_TEST))
+                    .compileOnly(dependency(GROUP_GOOGLE_CLOUD_FUNCTIONS, ARTIFACT_ID_FUNCTIONS_FRAMEWORK_API))
+                    .testImplementation(dependency(GROUP_GOOGLE_CLOUD_FUNCTIONS, ARTIFACT_ID_FUNCTIONS_FRAMEWORK_API))
+                    .testImplementation(micronautGcpDependency(ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP_TEST))
+                    .build();
+        } else if (runtime == MicronautRuntime.AZURE_FUNCTION) {
+            return Dependencies.builder()
+                    .implementation(micronautAzureDependency(ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP))
+                    .implementation("com.microsoft.azure.functions:azure-functions-java-library")
+                    .developmentOnly(micronautAzureDependency(ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP_TEST))
+                    .testImplementation(micronautAzureDependency(ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP_TEST))
+                    .build();
+
+        } else if (runtime == NETTY) {
+            return Dependencies.builder().implementation(dependency(GROUP_MICRONAUT, ARTIFACT_ID_MICRONAUT_SERVER_NETTY)).build();
+
+        } else if (runtime == TOMCAT) {
+            return Dependencies.builder().implementation(micronautServletDependency(ARTIFACT_ID_MICRONAUT_SERVLET_TOMCAT)).build();
+
+        } else if (runtime == JETTY) {
+            return Dependencies.builder().implementation(micronautServletDependency(ARTIFACT_ID_MICRONAUT_SERVLET_JETTY)).build();
+
+        } else if (runtime == UNDERTOW) {
+            return Dependencies.builder().implementation(micronautServletDependency(ARTIFACT_ID_MICRONAUT_SERVLET_UNDERTOW)).build();
+        }
```

Comment:
:o: I would add an `else if (runtime != NONE)` and throw here so that we know if we miss a branch in the future

Reply:
is runtime required? is NONE consider an invalid value?

Reply:
`NONE` is a valid value, it's a runtime which is "inject only". So we should check if it's _not_ none, in which case it means we have a missing branch.

Reply:
Resolved via: https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434/commits/00132ebfc90ed068282cab06c90baa0fdd325f5f